File: SDKs/Aspose.Words-Cloud-SDK-for-Java/src/main/java/com/aspose/words/model/TableProperties.java

package com.aspose.words.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class TableProperties {

    @SerializedName("Alignment")
    @Expose
    private String alignment;
    @SerializedName("AllowAutoFit")
    @Expose
    private Boolean allowAutoFit;
    @SerializedName("Bidi")
    @Expose
    private Boolean bidi;
    @SerializedName("BottomPadding")
    @Expose
    private Double bottomPadding;
    @SerializedName("CellSpacing")
    @Expose
    private Double cellSpacing;
    @SerializedName("LeftIndent")
    @Expose
    private Double leftIndent;
    @SerializedName("LeftPadding")
    @Expose
    private Double leftPadding;
    @SerializedName("PreferredWidth")
    @Expose
    private PreferredWidth preferredWidth;
    @SerializedName("RightPadding")
    @Expose
    private Double rightPadding;
    @SerializedName("StyleIdentifier")
    @Expose
    private String styleIdentifier;
    @SerializedName("StyleName")
    @Expose
    private String styleName;
    @SerializedName("StyleOptions")
    @Expose
    private String styleOptions;
    @SerializedName("TextWrapping")
    @Expose
    private String textWrapping;
    @SerializedName("TopPadding")
    @Expose
    private Integer topPadding;
    @SerializedName("link")
    @Expose
    private Link link;

    public String getAlignment() {
        return alignment;
    }

    public void setAlignment(String alignment) {
        this.alignment = alignment;
    }

    public Boolean getAllowAutoFit() {
        return allowAutoFit;
    }

    public void setAllowAutoFit(Boolean allowAutoFit) {
        this.allowAutoFit = allowAutoFit;
    }

    public Boolean getBidi() {
        return bidi;
    }

    public void setBidi(Boolean bidi) {
        this.bidi = bidi;
    }

    public Double getBottomPadding() {
        return bottomPadding;
    }

    public void setBottomPadding(Double bottomPadding) {
        this.bottomPadding = bottomPadding;
    }

    public Double getCellSpacing() {
        return cellSpacing;
    }

    public void setCellSpacing(Double cellSpacing) {
        this.cellSpacing = cellSpacing;
    }

    public Double getLeftIndent() {
        return leftIndent;
    }

    public void setLeftIndent(Double leftIndent) {
        this.leftIndent = leftIndent;
    }

    public Double getLeftPadding() {
        return leftPadding;
    }

    public void setLeftPadding(Double leftPadding) {
        this.leftPadding = leftPadding;
    }

    public PreferredWidth getPreferredWidth() {
        return preferredWidth;
    }

    public void setPreferredWidth(PreferredWidth preferredWidth) {
        this.preferredWidth = preferredWidth;
    }

    public Double getRightPadding() {
        return rightPadding;
    }

    public void setRightPadding(Double rightPadding) {
        this.rightPadding = rightPadding;
    }

    public String getStyleIdentifier() {
        return styleIdentifier;
    }

    public void setStyleIdentifier(String styleIdentifier) {
        this.styleIdentifier = styleIdentifier;
    }

    public String getStyleName() {
        return styleName;
    }

    public void setStyleName(String styleName) {
        this.styleName = styleName;
    }

    public String getStyleOptions() {
        return styleOptions;
    }

    public void setStyleOptions(String styleOptions) {
        this.styleOptions = styleOptions;
    }

    public String getTextWrapping() {
        return textWrapping;
    }

    public void setTextWrapping(String textWrapping) {
        this.textWrapping = textWrapping;
    }

    public Integer getTopPadding() {
        return topPadding;
    }

    public void setTopPadding(Integer topPadding) {
        this.topPadding = topPadding;
    }

    public Link getLink() {
        return link;
    }

    public void setLink(Link link) {
        this.link = link;
    }

}

File: SDKs/Aspose.Words-Cloud-SDK-for-Ruby/lib/aspose_words_cloud/models/table_properties.rb
module AsposeWordsCloud
  # 
  class TableProperties < BaseObject
    attr_accessor :alignment, :allow_auto_fit, :bidi, :bottom_padding, :cell_spacing, :left_indent, :left_padding, :preferred_width, :right_padding, :style_identifier, :style_name, :style_options, :text_wrapping, :top_padding, :link
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'alignment' => :'Alignment',
        
        # 
        :'allow_auto_fit' => :'AllowAutoFit',
        
        # 
        :'bidi' => :'Bidi',
        
        # 
        :'bottom_padding' => :'BottomPadding',
        
        # 
        :'cell_spacing' => :'CellSpacing',
        
        # 
        :'left_indent' => :'LeftIndent',
        
        # 
        :'left_padding' => :'LeftPadding',
        
        # 
        :'preferred_width' => :'PreferredWidth',
        
        # 
        :'right_padding' => :'RightPadding',
        
        # 
        :'style_identifier' => :'StyleIdentifier',
        
        # 
        :'style_name' => :'StyleName',
        
        # 
        :'style_options' => :'StyleOptions',
        
        # 
        :'text_wrapping' => :'TextWrapping',
        
        # 
        :'top_padding' => :'TopPadding',
        
        # 
        :'link' => :'link'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'alignment' => :'String',
        :'allow_auto_fit' => :'BOOLEAN',
        :'bidi' => :'BOOLEAN',
        :'bottom_padding' => :'Float',
        :'cell_spacing' => :'Float',
        :'left_indent' => :'Float',
        :'left_padding' => :'Float',
        :'preferred_width' => :'PreferredWidth',
        :'right_padding' => :'Float',
        :'style_identifier' => :'String',
        :'style_name' => :'String',
        :'style_options' => :'String',
        :'text_wrapping' => :'String',
        :'top_padding' => :'Float',
        :'link' => :'Link'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'Alignment']
        self.alignment = attributes[:'Alignment']
      end
      
      if attributes[:'AllowAutoFit']
        self.allow_auto_fit = attributes[:'AllowAutoFit']
      end
      
      if attributes[:'Bidi']
        self.bidi = attributes[:'Bidi']
      end
      
      if attributes[:'BottomPadding']
        self.bottom_padding = attributes[:'BottomPadding']
      end
      
      if attributes[:'CellSpacing']
        self.cell_spacing = attributes[:'CellSpacing']
      end
      
      if attributes[:'LeftIndent']
        self.left_indent = attributes[:'LeftIndent']
      end
      
      if attributes[:'LeftPadding']
        self.left_padding = attributes[:'LeftPadding']
      end
      
      if attributes[:'PreferredWidth']
        self.preferred_width = attributes[:'PreferredWidth']
      end
      
      if attributes[:'RightPadding']
        self.right_padding = attributes[:'RightPadding']
      end
      
      if attributes[:'StyleIdentifier']
        self.style_identifier = attributes[:'StyleIdentifier']
      end
      
      if attributes[:'StyleName']
        self.style_name = attributes[:'StyleName']
      end
      
      if attributes[:'StyleOptions']
        self.style_options = attributes[:'StyleOptions']
      end
      
      if attributes[:'TextWrapping']
        self.text_wrapping = attributes[:'TextWrapping']
      end
      
      if attributes[:'TopPadding']
        self.top_padding = attributes[:'TopPadding']
      end
      
      if attributes[:'link']
        self.link = attributes[:'link']
      end
      
    end

  end
end
